src/monattack.cpp

```diff
@@ -5801,8 +5801,12 @@ bool mattack::doot( monster *z )
 
 bool mattack::dodge_check( monster *z, Creature *target )
 {
+    const float dodge_ability = target->get_dodge();
+    if( dodge_ability > 0.0 ) {
+        target->on_try_dodge();
+    }
     ///\EFFECT_DODGE increases chance of dodging, vs their melee skill
-    float dodge = std::max( target->get_dodge() - rng( 0, z->get_hit() ), 0.0f );
+    float dodge = std::max( dodge_ability - rng( 0, z->get_hit() ), 0.0f );
```

Comment:
```suggestion
    float attack_roll = z->get_hit() + rng_normal( 0, 5 );
```
Not strictly necessary but I think putting this on a separate line makes it easier to parse and edit later.